File: testsuite/OfflineExamples/RaytracingOfflineExample/src/RaytracerSerialExecutor.java
import vsdk.toolkit.common.RendererConfiguration;
import vsdk.toolkit.environment.scene.SimpleSceneSnapshot;
import vsdk.toolkit.gui.feedback.ProgressMonitor;
import vsdk.toolkit.media.RGBImageUncompressed;
import vsdk.toolkit.render.SimpleRaytracer;

final class RaytracerSerialExecutor implements RaytracerExecutor {
    @Override
    public void run(SimpleRaytracer visualizationEngine,
                    RGBImageUncompressed resultingImage,
                    RendererConfiguration rendererConfiguration,
                    SimpleSceneSnapshot sceneSnapshot,
                    ProgressMonitor reporter)
    {
        visualizationEngine.execute(
            resultingImage,
            rendererConfiguration,
            sceneSnapshot,
            reporter,
            null);
    }
}

File: typescript/testsuite/OfflineExamples/RaytracingOfflineExample/src/RaytracerSerialExecutor.ts
import type {
    ProgressMonitor,
    RGBImageUncompressed,
    RendererConfiguration,
    SimpleRaytracer,
    SimpleSceneSnapshot,
} from "@vitral/base";
import type { RaytracerExecutor } from "./RaytracerExecutor.js";

export class RaytracerSerialExecutor implements RaytracerExecutor {
    public run(
        visualizationEngine: SimpleRaytracer,
        resultingImage: RGBImageUncompressed,
        rendererConfiguration: RendererConfiguration,
        sceneSnapshot: SimpleSceneSnapshot,
        reporter: ProgressMonitor,
    ): void {
        visualizationEngine.execute(resultingImage, rendererConfiguration, sceneSnapshot, reporter, null);
    }
}
